### ISSUE TITLE
Please add samples 

### ISSUE THREAD

**Issue by rajjaiswalsaumya** [leMaik/swing-material#3]
![issue](https://cloud.githubusercontent.com/assets/5713776/18576553/35023aac-7c00-11e6-8668-6e694619388f.PNG)
Please provide samples on how to use this library. It is pretty great solution and I think this should be an active community. 

Suggestion -> 
please change your project structure as ->

```
swing-material
         swing-material
         swing-material-samples
```

This way anyone will be able to use it by looking into samples and also you will be able to run samples as live tests upon build.

My issue, I am adding materialpasswordfield but the hint label is painting below box line. Help!!

**Comment by leMaik** [comment]
I will add samples and better documentation soon, I just didn't think that anyone would have a use case for this project. Meanwhile, you may want to take a look at the [Craften Launcher](https://github.com/TeamWertarbyte/craften-launcher) it uses swing-material.

I think that you have to specify the height of the password field. The project mentioned above uses a password field.

**Comment by rajjaiswalsaumya** [comment]
I can see you have not acknowledged layout managers and you have set all bounds manually which wont be a good idea in case of differing resolution screen. Can you just provide me a time so that we could discuss about it ?

**Comment by rajjaiswalsaumya** [comment]
Btw I have set dimensions of the password field but I m using flowlayout for the panel. I think with that they are not compatible.

**Comment by leMaik** [comment]
@rajjaiswalsaumya Indeed, I set all bounds individually, but it seems to work fine on different screens. I don't know how this could be used with `FlowLayout`, but I'm sure that it should be possible by specifying the `defaultSize` of the Material components.

**Comment by DragShot** [comment]
Well, I understand uses of this library are sort of few at the moment, as the suite of components at our disposition is small and in certain aspects incomplete; but at least let's appreciate the shadows, elevation handling and ripples, as they are certainly very well done. One or two optimizations could be done though, my old piece of PC could barely deal with the continuous shadow processing plus the rest of animations.

I'm trying to make an use out of it because I was looking for something like this yesterday for a small, sort of e-book reader. I'll try to fill in some blanks and tweak one or two things while I'm on it, and if things go fine then I'll fork this in order to send a pull request with the changes made.

![swing_material_001.png](https://i58.servimg.com/u/f58/17/37/76/58/swing_10.png)

**Comment by leMaik** [comment]
@DragShot That would be great! :+1: 
Unfortunately, I don't have much time right now to document this library, but I'll come back to this when I have some more spare time. I know it would be much easier to use this if there were a few examples. Well, at least, we have Javadoc everywhere.

Edit: To be honest, this library contains exactly the set of components that I needed in our [Minecraft launcher](https://github.com/TeamWertarbyte/craften-launcher). I don't use this library for other projects and nobody has requested more components yet. :sweat_smile:

**Comment by DragShot** [comment]
I've forked your project and added some of the changes I've made on my end so far. There are inline comments starting with "DS-addons:" explaining the purpose of each modification. Also, some docs have been added.

You can [take a look](https://github.com/DragShot/swing-material) and tell me what you think.

**Comment by leMaik** [comment]
@DragShot Wow, that are some great additions! I'd really appreciate a PR from you, just remove the inline comments (you can see the changes in the commit diff). :+1:

**Comment by DragShot** [comment]
@leMaik Yes, I am aware of the diff, those comments are only for preview and will be removed before the PR is sent. Right now I'm working in a sort of Frame decorator (title bar, control buttons, window resizing) for the MaterialWindow. I'll commit the changes once it is done.

**Comment by leMaik** [comment]
@DragShot It would be easier for me to review the changes and apply them one by one if you'd create a full request for every change / new feature. :wink:

**Comment by DragShot** [comment]
Alright, I've cleaned the files and posted the first PE. You should be able to see it now, @leMaik.

**Comment by leMaik** [comment]
@pedrohidalgo Ping :wink:
